app/server/utils/user.py

```diff
@@ -159,7 +160,6 @@ def update_transfer_account_user(user,
 
     # remove all roles before updating
     user.remove_all_held_roles()
-    flag_modified(user, '_held_roles')
```

Comment:
flag_modified breaks `load_history().has_changes()`, so I had to cut it out of `user_roles` logic. Everything still works as normal though!